### PR TITLE
0.13 fix exec to not deploy anything

### PR DESCRIPTION
**Pull request by vvagaytsev**
**What this PR does / why we need it**:
This prevents the `garden exec` command from executing any deployments. If the target container is found, the command execution proceeds. Otherwise, an error is thrown.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
The commits will be squashed before the merge.

### CODE REVIEW

File: core/src/commands/exec.ts [hunk 1]
```diff
@@ -62,29 +63,57 @@ export class ExecCommand extends Command<Args, Opts> {
...
+    const status = await router.deploy.getStatus({ action: resolved, graph, log })
+    const deployState = status.result.detail?.state
+    switch (deployState) {
+      // Warn if the deployment is not ready yet or unhealthy, but still proceed.
+      case undefined:
+      case "deploying":
+      case "outdated":
+      case "unhealthy":
+      case "unknown":
+        log.warn(chalk.white(`Current state: ${chalk.whiteBright(deployState)}`))
```
Comment (edvald) [inline core/src/commands/exec.ts:95]: This log message could be clearer. Suggest making it yellow+bold and say something like `\`The current state of ${action.key()} is ${chalk.whiteBright(deployState)}. If this command fails, you may need to re-deploy it with the ${chalk.whiteBright("deploy")} command\``.

File: core/src/commands/exec.ts [hunk 2]
```diff
@@ -62,29 +63,57 @@ export class ExecCommand extends Command<Args, Opts> {
...
+        break
+      // Only fail if the deployment is missing or stopped.
+      case "missing":
+      case "stopped":
+        throw new NotFoundError(
+          `Cannot execute command in the '${action.name}' service. The target container is ${deployState}.`,
```
Comment (edvald) [inline core/src/commands/exec.ts:101]:
```suggestion
          `${action.key()} status is ${deployState}. Cannot execute command.`,
```

File: core/src/plugin/handlers/Deploy/exec.ts [hunk 3]
```diff
@@ -35,7 +35,7 @@ export const execInDeployResultSchema = () =>
 
 export class ExecInDeploy<T extends DeployAction = DeployAction> extends ActionTypeHandlerSpec<
   "Deploy",
-  ExecInDeployParams<Executed<T>>,
+  ExecInDeployParams<Resolved<T>>,
```
Comment (edvald) [inline core/src/plugin/handlers/Deploy/exec.ts:38]: I don't think this makes sense. The exec plugin handlers will rely on a status being present on the action, which means it needs to be executed. We can approach this differently. I'll add a commit with my suggested approach.

Comment (edvald) [reply core/src/plugin/handlers/Deploy/exec.ts]: I've added a commit that you can check out and then squash in if it looks good @vvagaytsev

Comment (vvagaytsev) [reply core/src/plugin/handlers/Deploy/exec.ts]: Looks good, thank you!

Comment (vvagaytsev) [reply core/src/plugin/handlers/Deploy/exec.ts]: And let's approve this first :)

**Review by thsig — APPROVED** [review]

**Pull request merged**